frontend/src/views/Game.tsx

```diff
@@ -47,6 +54,65 @@ function GamePage() {
    */
   useBeforeunload(() => 'Leaving this page may cause you to lose your current code and data.');
 
+  // Calculate and set the new clock on the frontend.
+  const calculateSetClock = useCallback((gameTimerParam: GameTimer) => {
```

Comment:
Is it possible to move this to a separate file in like a helper method? Or maybe create a new component like GameClock or something that takes in the gameTimerParam + an end game callback as props and then handles all the counting down stuff in that class

Reply:
Yeah I intended to create a new component at some point -- I can do that now!

Reply:
Done! I will note that when I was doing this and testing the result, it seems like on the game page the socket is being subscribed to twice -- I'm not sure why this is, I don't think it's because of any change I made here, but there does seem to be an error somehow.

Reply:
I also want to note that there is a brief moment while the timer shows "00:00" before the `GameTimer` is transmitted through. The timer then starts about 5 seconds after the actual timer amount; so if the timer is 15:00, the first time you usually see is 14:55. If this is an issue, let me know 🙂

Reply:
I fixed the issue with the double-subscribed socket (I added a dependency in a `useEffect` call that was not necessary). 👍